hub/api/dataset.py

```diff
@@ -73,11 +75,16 @@ def __init__(
             creds = {}
         base_storage = get_storage_provider(path, storage, read_only, creds, token)
 
-        # done instead of directly assigning read_only as backend might return read_only permissions
-        if hasattr(base_storage, "read_only") and base_storage.read_only:
+        if read_only:
```

Comment:
this logic was already handled within the get_storage_provider call.
it would already set the base storage read_only equal to true if the credentials gave read_only access or the user specified the ready only mode.
this change seems unnecessary.

Reply:
it is not handled, in some cases it does not do this.

Reply:
the readonly tests i wrote fail when this isn't here

Reply:
actually, i just removed this and re-ran, i guess it doesn't fail? weird. will remove this